### PR TITLE
Limit marker types

### PR DESCRIPTION
**Pull request by RealOrangeOne**
The aruco types had a suffix based on how many markers they would capture. Whilst OpenCV has these, we really don't need to expose them. Simply exposing the max allowed covers the bases, and will reduce confusion.

This also adds some properties onto the marker type to get its dictionary size, number of pixels, max id etc.

Tests have been improved slightly. Rather than randomly using a max id, we use the smallest `max_id` that a marker allows, to fully test as much as possible.

### CODE REVIEW

File: zoloto/marker_type.py [hunk 1]
```diff
@@ -1,39 +1,59 @@
...
+
+    @property
+    def min_marker_size(self) -> int:
+        """
+        Minimum size of a marker in pixels
+        """
+        return self.marker_bits + 2
+
+    @property
+    def marker_bits(self) -> int:
```
Comment (RealOrangeOne) [inline zoloto/marker_type.py:43]: Not super happy with the naming here. eg `ARUCO_7X7` creates a 7x7 marker, so this produces 7.

File: zoloto/marker_type.py [hunk 2]
```diff
@@ -1,39 +1,59 @@
...
+
+    @property
+    def max_id(self) -> int:
+        """
+        The highest id available
+        """
+        return self.marker_count - 1
+
+    @property
+    def min_marker_size(self) -> int:
```
Comment (RealOrangeOne) [inline zoloto/marker_type.py:36]: Not super happy with the naming here. This is the size just on 1 axis

Comment (trickeydan) [reply zoloto/marker_type.py:36]: Looks like the aruco paper refers to this as size.

File: zoloto/marker_type.py (hunk 1 above)
Comment (trickeydan) [reply zoloto/marker_type.py:43]: `marker_array_size`?

File: zoloto/marker_type.py (hunk 2 above)
Comment (raccube) [reply zoloto/marker_type.py:36]: I think marker size is the only thing that makes any sense here

**Pull request merged**